Jetchat/app/src/main/java/com/example/compose/jetchat/components/JetchatScaffold.kt

```diff
@@ -43,3 +96,70 @@ fun JetchatScaffold(
         )
     }
 }
+
+@Composable
+fun TopBarScaffold(
+    scaffoldState: ScaffoldState = rememberScaffoldState(),
+    drawerState: DrawerState,
+    complexTopBar: Boolean,
+    content: @Composable (PaddingValues) -> Unit
+) {
+    var functionalityNotAvailablePopupShown by remember { mutableStateOf(false) }
+    val scope = rememberCoroutineScope()
+    val onNavIconClicked: () -> Unit = {
+        scope.launch {
+            drawerState.open()
+        }
+    }
+
+    JetchatTheme {
+        Scaffold(
+            scaffoldState = scaffoldState,
+            topBar = {
+                if (complexTopBar) {
+                    ChannelNameBar(
+                        channelName = exampleUiState.channelName,
+                        channelMembers = exampleUiState.channelMembers,
+                        onNavIconPressed = onNavIconClicked,
+                        // Use statusBarsPadding() to move the app bar content below the status bar
+                        modifier = Modifier.statusBarsPadding(),
+                    )
+                } else {
+                    JetchatAppBar(
+                        // Use statusBarsPadding() to move the app bar content below the status bar
+                        modifier = Modifier
+                            .fillMaxWidth()
+                            .statusBarsPadding(),
+                        onNavIconPressed = onNavIconClicked,
+                        title = { },
+                        actions = {
+                            CompositionLocalProvider(
+                                LocalContentAlpha provides ContentAlpha.medium
+                            ) {
+                                // More icon
+                                Icon(
+                                    imageVector = Icons.Outlined.MoreVert,
+                                    modifier = Modifier
+                                        .clickable(
+                                            onClick = {
+                                                functionalityNotAvailablePopupShown = true
+                                            }
+                                        )
+                                        .padding(horizontal = 12.dp, vertical = 16.dp)
+                                        .height(24.dp),
+                                    contentDescription = stringResource(id = R.string.more_options)
+                                )
+                            }
+                        }
+                    )
+                }
+            },
+            content = {
+                if (functionalityNotAvailablePopupShown) {
+                    FunctionalityNotAvailablePopup { functionalityNotAvailablePopupShown = false }
+                }
+                content(it)
+            }
+        )
+    }
+}
```

Comment:
nit; Can you make sure your files have new-lines at the end?

Reply:
Done